documentation/design_proposal/002_grafana_v1beta1.md

```diff
@@ -0,0 +1,191 @@
+# Design proposal 002 grafana CRD v1beta1
+
+## Summary
+
+Update the grafana CRD to be easier to use and provide more customization opportunities.
+
+## Info
+
+status: Draft
+
+## Motivation
+
+More and more users have asked us for more possibility to customize the grafana deployment but also
+other objects that grafana-operator manages.
+
+The new version of the CRD should provide the entire deployment/containers object when possible while at the same time
+be opinionated how the default object that we generate should look like just like even more then we do today.
+
+The grafana CRD API is currently a bit sprawling and it can be hard to find how to
+overwrite a number of default values, especially when it comes to deployment configuration.
+
+As a part of the v1beta1 release we want to make this better and at the same time
+give our users more power to customize there deployments.
```

Comment:
This would be a breaking change right?

Reply:
Added a note in the summary to make it super clear that this design idea contains allot of breaking changes.